cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,264 @@
+
+### Non-goals
+
+* **Azure Managed Identity support**
+    - Managed Identity support for AKS is a much older feature, and Workload Identity is the replacement. We're not planning to support Managed Identity, as any user should be able to upgrade to Workload Identity.
+
+* **AWS Workload Identity (IRSA) support**
+    - Will be tracked as separate work: https://github.com/radius-project/radius/issues/7618
+
+### User scenarios
+
+#### User story 1: As an infrastructure operator, I can configure Radius to use Azure workload identity via interactive experience.
+
+1. Install the `rad` CLI.
+
+1. run `rad init --full` to begin the interactive experience
+    ```
+    rad init --full
+    ```
+
+1. Enter the environment name, namespace and select "yes" to add a cloud provider and select "Azure" to add an Azure cloud provider.
+    ```
+    Select your cloud provider:
+    > 1. Azure
+      2. AWS
+      3. [back]
+    ```
+
+1. Enter or use the Azure subscription and resource group name. Select  "Workload Identity" for the Azure cloud provider.
+    ```
+    Select the identity for the Azure cloud provider 
+    1. Service principal                               
+    > 2. Workload Identity    
+    ```
+
+1. Enter the Entra ID Application Client ID and Tenant ID.
+    ```
+    Please follow the guidance at aka.ms/rad-workload-identity to set up workload identity for Radius.
+
+    Enter the Client ID of the Entra ID Application that you have created for Radius.
+
+    > Enter the Client ID:
+
+    Enter the Tenant ID of the Entra ID Application that you have created for Radius.
+
+    > Enter the Tenant ID:
+    ```
+
+Radius is now configured to use Azure workload identity for authentication.
+
+#### User story 1: As an infrastructure operator, I can configure Radius to use Azure workload identity without an interactive experience.
+
+1. Install the `rad` CLI.
+
+1. Install Radius using `rad install kubernetes`
+    ```
+    rad install kubernetes --set global.azureWorkloadIdentity.enabled=true
+    ```
+
+1. Create a Radius environment and add an Azure cloud provider.
+    ```
+    rad group create default
+    rad env create default
+    rad env update default --azure-subscription-id <subscription-id> --azure-resource-group <resource-group> 
+    ```
+
+1. Register the Entra ID Application using `rad credential register azure wi`.
+    ```
+    rad credential register azure wi --client-id <client-id> --tenant-id <tenant-id>
+    ```
+
+Radius is now configured to use Azure workload identity for authentication.
+
+## User Experience
+
+We will be adding a new option to the `rad init --full` command to allow the user to choose between Azure service principal identity and Azure workload identity.
+
+We will be adding a new option to the Radius Helm chart to allow the user to enable Azure Workload Identity during installation via `rad install kubernetes --set global.azureWorkloadIdentity.enabled=true`.
+
```

Comment:
would we want a informational message to user to follow up with a use rad credentials to complete the setup message?